tamr_toolbox/filesystem/cloud.py

```diff
@@ -0,0 +1,83 @@
+import os
+
+# Building our documentation requires access to all dependencies, including optional ones
+# This environments variable is set automatically when `invoke docs` is used
+BUILDING_DOCS = os.environ.get("TAMR_TOOLBOX_DOCS") == "1"
+if BUILDING_DOCS:
+    # Import relevant optional dependencies
+    from google.cloud import storage
+    from mypy_boto3_s3.client import S3Client
+
+
+def gcs_upload(
+    cloud_client: "storage.Client",
+    source_filepath="path_to_my_local_file",
+    destination_filepath="path_to_my_file_in_bucket",
+    bucket_name="sample_bucket",
```

Comment:
These parameters still need type hinting and should not have default values.  It would also be best to force their use as keyword-only by using the `*` parameter separator:
```suggestion
    *,
    source_filepath: str,
    destination_filepath: str,
    bucket_name: str,
```